base/timing.jl

```diff
@@ -207,9 +217,10 @@ macro time(ex)
         local stats = gc_num()
         local elapsedtime = time_ns()
         local compile_elapsedtime = cumulative_compile_time_ns_before()
-        local val = $(esc(ex))
-        compile_elapsedtime = cumulative_compile_time_ns_after() - compile_elapsedtime
-        elapsedtime = time_ns() - elapsedtime
+        local val = @__tryfinally($(esc(ex)),
+            (elapsedtime = time_ns() - elapsedtime;
+            compile_elapsedtime = cumulative_compile_time_ns_after() - compile_elapsedtime)
```

Comment:
Should the order of these be switched to avoid the possibility of compile time being more than 100%?

Reply:
Ok, Oscar made the point that we care more about an accurate elapsed time than about compile time fraction, so maybe it's ok.